Validation/MtdValidation/plugins/BtlDigiHitsValidation.cc

```diff
@@ -0,0 +1,279 @@
+// -*- C++ -*-
+//
+// Package:    Validation/MtdValidation
+// Class:      BtlDigiHitsValidation
+//
+/**\class BtlDigiHitsValidation BtlDigiHitsValidation.cc Validation/MtdValidation/plugins/BtlDigiHitsValidation.cc
+
+ Description: BTL DIGI hits validation
+
+ Implementation:
+     [Notes on implementation]
+*/
+
+#include <string>
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include "DQMServices/Core/interface/DQMEDAnalyzer.h"
+#include "DQMServices/Core/interface/MonitorElement.h"
+
+#include "DataFormats/FTLDigi/interface/FTLDigiCollections.h"
+
+#include "DataFormats/ForwardDetId/interface/BTLDetId.h"
+
+#include "Geometry/Records/interface/MTDDigiGeometryRecord.h"
+#include "Geometry/Records/interface/MTDTopologyRcd.h"
+#include "Geometry/MTDGeometryBuilder/interface/MTDGeometry.h"
+#include "Geometry/MTDNumberingBuilder/interface/MTDTopology.h"
+
+#include "Geometry/MTDGeometryBuilder/interface/ProxyMTDTopology.h"
+#include "Geometry/MTDGeometryBuilder/interface/RectangularMTDTopology.h"
+
+
+class BtlDigiHitsValidation : public DQMEDAnalyzer {
+
+public:
+  explicit BtlDigiHitsValidation(const edm::ParameterSet&);
+  ~BtlDigiHitsValidation() override;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+
+private:
+  void bookHistograms(DQMStore::IBooker &,
+		      edm::Run const&,
+		      edm::EventSetup const&) override;
+
+  void analyze(const edm::Event&, const edm::EventSetup&) override;
+
+  // ------------ member data ------------
+
+  const MTDGeometry* geom_;
+  const MTDTopology* topo_;
```

Comment:
Seems that these don't have to be members.

Reply:
Done